core/camel-support/src/main/java/org/apache/camel/support/DefaultComponent.java

```diff
@@ -551,6 +551,9 @@ protected boolean useIntrospectionOnEndpoint() {
      * @throws IllegalArgumentException if referenced object was not found in registry.
      */
     public <T> T resolveAndRemoveReferenceParameter(Map<String, Object> parameters, String key, Class<T> type, T defaultValue) {
+        if (parameters.containsKey(key) && type.isInstance(parameters.get(key).getClass())) {
```

Comment:
Parameter value can be null. Actually many tests are failing with `NullPointerException`.

Reply:
Should be OK now. Thanks.

Reply:
No. `parameters` will be never null. But value in map (`parameters.get(key)`) can be null. The NPE is still there. You can rollback last commit and instead replace `parameters.containsKey(key)` with `parameters.get(key) != null`

Reply:
Done, thanks